### ISSUE TITLE
Encryption passphrase is showing immediately

### ISSUE THREAD

**Issue by tobiasKaminsky** [nextcloud/desktop#155]
- a new user on server
- new test user on laptop
- first start of client & connecting to new server user
- shows immediately the encryption passphrase dialog

--> this should only be shown once an user creates an encrypted folder

**Comment by tcanabrava** [comment]
"Why?" - The server has the capability of encrypting files, *if* I show it on the first folder encryption the user migth think wrongly that is the *folder* password and not the encryption mechanism password.

**Comment by tobiasKaminsky** [comment]
Maybe there are uses that do not want to use E2E at all and then click away this message. And after some time they want to try the feature and they do not have the passphrase any more.

I would show it once they access an encrypted folder or once the user creates an encrypted folder and make sure that they understand that the passphrase is for the entire e2e system.

**Comment by tobiasKaminsky** [comment]
Maybe @jancborchardt as we discussed this part on android?

**Comment by tcanabrava** [comment]
If the user dismisses the dialog and doesn't copy he didn't actualy lost anything as the key will be stored in the keychain and he can retrieve it later. perhaps we can specify that somewhere.

**Comment by jancborchardt** [comment]
I can’t find the issue where we discussed this at the moment, but as @tobiasKaminsky said, we should do this here exactly as discussed on Android. Any talk of encryption should only start as soon as you choose to encrypt a folder, or decrypt an existing one. No dialog which comes up immediately after setup.

**Comment by rullzer** [comment]
We actually follow the RFC here... https://github.com/nextcloud/end_to_end_encryption_rfc/blob/master/RFC.md#initial-device

```
When a device is initially added to an account the device has to generate all relevant key material for the user account and provision those on the server.
```

I agree the UX is not optimal. But this is currently very low on my list of things to fix for the first test version ;)

**Comment by jancborchardt** [comment]
Then the RFC is not written with the user in mind. ;)

And this would be in the release, right? So not a first test version. This would impair the default UX of everyone installing the client.

**Comment by tobiasKaminsky** [comment]
> We actually follow the RFC here... 

Then we should change this. As said, there is no point in forcing an user to have an e2e passphrase if he will never ever use this. Imagine that it is enabled on a big instance (>10k users) and only a few want to use it...but all have to setup e2e?

**Comment by tobiasKaminsky** [comment]
> I can’t find the issue where we discussed this at the moment

I think this was at the conf and I put our discussion then in some meta - issue.

**Comment by rullzer** [comment]
> Then we should change this. As said, there is no point in forcing an user to have an e2e passphrase if he will never ever use this. Imagine that it is enabled on a big instance (>10k users) and only a few want to use it...but all have to setup e2e?

Yes yes, I get the bad UX and the trouble. But still the client in its current form follows the specification. The change is not so small. So that is why I said that it isn't high on my list of priorities. Especially since we are not near anything that is a stable release.

> I think this was at the conf and I put our discussion then in some meta - issue.

So not to surprising we didn't find it then :wink:

**Comment by camilasan** [comment]
I would not tag it as bug since it is working as expected but we can definitely enhance the user experience.

**Comment by er-vin** [comment]
Hello,

I think this issue is properly addressed with PR #1241 so closing now.

Regards.
